enhancements/sdk-supports-operands-deletion.md

```diff
@@ -0,0 +1,135 @@
+
+Any operator that deletes or expects delete events for CRs that create resources can use finalizers or owner references, although the latter can be used when those CRs only create resources in the same namespace. Additionally finalizers permit customizable deletion steps, ex. update a CR's status during deletion with a particular message. Because finalizers are broadly applicable and give operator authors control over how CR deletion proceeds, operator projects would benefit from some initial finalizer code scaffolded by default.
+
+To enable this, it would be ideal to have a finalizer "no-op" handler, i.e. some code that does not clean up a CR but has a `TODO(user)` comment, scaffolded by default in the kubebuilder project. The idea is to always scaffold out a finalizer, a delete event handler and a Reconciler that checks for Finalizers and has a no-op in cases where it's not required to use a Finalizer.
+
+The Reconciler checks for presence of a Finalizer in a CR object. If the CR object is marked for deletion, indicated by `metadata.deletionTimestamp` being set, the Reconciler runs the finilizer logic and removes the finalizer. If the finalization logic fails for some reason, then the Reconciler keeps the finalizer to retry during the next reconcilation. Typically, the finalizer logic would entail necessary cleanup steps that the operator needs to perform before the CR can be deleted. See [finalizer example][finalizer-example] for more details.
+
+#### Configuration of the cleanup API in SDK
+
+A [ClusterServiceVersion (CSV)][csv] represents a particular version of a running operator on a cluster and is an actual installation of the operator. The CSV will support the `spec.cleanup` field once the [operand cleanup proposal][operand-cleanup-proposal] is implemented
+
+To enable configuration of operand deletion on the CSV, the `operator-sdk generate kustomize manifests` command will generate CSVs with the `spec.cleanup` field set to `false` by default. The user can then explicitly set this field to true to opt-in to automatic deletion of CRs using OLM.
+
+#### Existing cleanup command in SDK
+
+The existing `operator-sdk cleanup` command currently deletes the CRDs and all other resources owned by the operator, without an option to disable this behavior. Ideally, the cleanup command should have certain flags that prevent deletion of CRDs, for feature parity with the [kubectl operator plugin](https://github.com/operator-framework/kubectl-operator/).
+
+Existing cleanup command in OSDK:
+
+```
+operator-sdk cleanup <operatorPackageName>
+```
+
+Proposed cleanup command with flags:
+
+```
+operator-sdk cleanup <operatorPackageName> --delete-crds --delete-all
+```
+
+`--delete-crds` boolean flag should be added to the `cleanup` command, and defaulted to `true` to preserve current behavior. When `--delete-crds=false`, the command will not clean up CRDs.
```

Comment:
I don't think it can be done with cobra. Rethinking the flags, I think `--delete-all` is kind of redundant because of today the cleanup command removes everything. I removed this flag from the EP and now only have `[--delete-crds]` flag which will be set to false by default to preserve the current behavior of the cleanup command. 

If it is set to true by the user, then OSDK will not remove any CRDs.